src/components/Dashboard/Dashboard.tsx

```diff
@@ -13,26 +15,29 @@ import { ToastProvider } from "../../context/toasts";
 // Constants
 import { ACCOUNT_TAB, ADMIN_TAB, ENODE_TAB } from "../../constants/tabs";
 
-const Dashboard = ({ tab, setTab, dataReady }) => (
+const Dashboard: React.FC<any> = ({ tab, setTab }) => (
     <Fragment>
         <TabSelector setTab={setTab} tab={tab} />
-        {!dataReady ? (
-            <LoadingPage />
-        ) : (
+        {
             <ToastProvider>
                 <Toasts />
-                <AccountTab isOpen={tab === ACCOUNT_TAB} />
-                <AdminTab isOpen={tab === ADMIN_TAB} />
-                <EnodeTab isOpen={tab === ENODE_TAB} />
+                <AdminDataProvider>
+                    <AccountDataProvider>
+                        <AccountTab isOpen={tab === ACCOUNT_TAB} />
+                    </AccountDataProvider>
+                    <AdminTab isOpen={tab === ADMIN_TAB} />
+                    <NodeDataProvider>
+                        <EnodeTab isOpen={tab === ENODE_TAB} />
+                    </NodeDataProvider>
+                </AdminDataProvider>
```

Comment:
I am assuming it is by design? i.e. AccountDataProvider is nested under AdminDataProvider?

Reply:
yeah, admin data is need in all 3 tabs to see if we're the admin, account & node data is only needed in their respective tabs